packages/libraries/yoga/src/index.ts

```diff
@@ -124,10 +116,12 @@ export function useHive(clientOrOptions: HiveClient | HivePluginOptions): Plugin
               errors.push(...ctx.result.errors);
             },
             onEnd() {
-              void record.callback(
-                args,
-                errors.length ? { errors } : {},
-                record.experimental__documentId,
+              args.contextValue.waitUntil(
+                record.callback(
+                  args,
+                  errors.length ? { errors } : {},
+                  record.experimental__documentId,
+                ),
```

Comment:
On Cloudflare Workers, I know what this will do (suspend the worker from being terminated until the promise resolves).

Can you explain to me what implications this has within a Node.js environment? Is `waitUntil` there just a dummy implementation?

Reply:
No with this PR `waitUntil` has been implemented in Node.js fully; https://github.com/ardatan/whatwg-node/pull/1830
All the promises passed to `waitUntil` are kept until they get resolved, and disposal waits for it.
Disposal logic has also been added with this PR so any disposable plugins together with ongoing waitUntil promises are resolved through this disposal logic.
In Node.js and compatible environments, we add a listener for process termination for this disposal logic.

Reply:
```ts
const deferredForWaitUntil = createDeferred();
const deferredForPlugin = createDeferred();
const yoga = createYoga({
  schema: createSchema({
     // ...
     resolvers: {
        Query: {
            someStuff(_, __, ctx) {
                ctx.waitUntil(
                   deferredForWaitUntil.promise.then(() => {
                     console.log('Wait until')
                   })
                )
            }
        }
     }
  })
  plugins: [
     {
       [Symbol.asyncDispose]() {
          console.log('Plugin disposal')
          return deferredForPlugin.promise;
       }
     }
  ]
}).

const res = await yoga.fetch('/graphql');
console.log(await res.text()) // Not blocked and printed

// await using is better but I want to show the timing
// This is also called when Node.js process terminates
yoga[Symbol.asyncDispose]().then(() => {
  console.log('Fully disposed');
})

deferredForWaitUntil.resolve() // Printed waitUntil but stil lwaiting for promise
deferredForPlugin.resolve() // Printed Plugin disposal and Fully disposed
```

Reply:
Did improvements based on @n1ru4l 's feedback!
https://github.com/ardatan/whatwg-node/pull/1899

Reply:
@ardatan I'm completely clueless about the whole `waitUntil` and `disposble` stuff
where can I read and learn about it?

Reply:
This is from my other PR adding documentation about this logic;
This is the section I added to our documentation -> https://github.com/dotansimha/graphql-yoga/blob/4743ee974ed7decba8d89b26251e214d7cf42885/website/src/pages/docs/features/explicit-resource-management.mdx

This is the PR -> https://github.com/dotansimha/graphql-yoga/pull/3567